compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/IncrementalCompilerRunner.kt

```diff
@@ -75,192 +86,210 @@ abstract class IncrementalCompilerRunner<
         providedChangedFiles: ChangedFiles?,
         projectDir: File? = null
     ): ExitCode = reporter.measure(BuildTime.INCREMENTAL_COMPILATION_DAEMON) {
-        try {
-            compileImpl(allSourceFiles, args, messageCollector, providedChangedFiles, projectDir)
-        } finally {
-            reporter.measure(BuildTime.CALCULATE_OUTPUT_SIZE) {
-                reporter.addMetric(
-                    BuildPerformanceMetric.SNAPSHOT_SIZE,
-                    buildHistoryFile.length() + lastBuildInfoFile.length() + abiSnapshotFile.length()
+        return when (val result = tryCompileIncrementally(allSourceFiles, providedChangedFiles, args, projectDir, messageCollector)) {
+            is ICResult.Completed -> {
+                reporter.debug { "Incremental compilation completed" }
+                result.exitCode
+            }
+            is ICResult.RequiresRebuild -> {
+                reporter.info { "Non-incremental compilation will be performed: ${result.reason}" }
+                reporter.addAttribute(result.reason)
+
+                compileNonIncrementally(
+                    result.reason, allSourceFiles, args, projectDir, trackChangedFiles = providedChangedFiles == null, messageCollector
                 )
-                if (cacheDirectory.exists() && cacheDirectory.isDirectory()) {
-                    cacheDirectory.walkTopDown().filter { it.isFile }.map { it.length() }.sum().let {
-                        reporter.addMetric(BuildPerformanceMetric.CACHE_DIRECTORY_SIZE, it)
-                    }
+            }
+            is ICResult.Failed -> {
+                reporter.warn {
+                    // The indentation after the first line is to make it clear that all of this is one message and is different from the
+                    // next message that appears in the log.
+                    """
```

Comment:
Yeah I think it's verbose too. The reason is that if we write the error message like this:

```
"""
|abcdef
|abcdef
|abcdef
""".trimMargin()
```

In the log, it would look like this:
```
[Kotlin][IC] abcdef
abcdef
abcdef
xyzxyz
xyzxyz
```

I think it's better if the log looks like this:

```
[Kotlin][IC] abcdef
    abcdef
    abcdef
xyzxyz
xyzxyz
```

That's why I added the indentation and added a comment to explain why.

I just updated the comment to make it a bit shorter:
```
// The indentation after the first line is intentional (so that this message is distinct from next message)
```

(I'll remove it if you still think it's not important.)